clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4536,23 +4548,162 @@ void SYCLIntegrationFooter::emitSpecIDName(raw_ostream &O, const VarDecl *VD) {
   O << "";
 }
 
+template <typename Before, typename After>
+static void PrintNSHelper(Before B, After A, raw_ostream &O,
+                          const DeclContext *DC) {
```

Comment:
I am not sure if my suggestion is an improvement, but I would like to see something other than single-letter variable names for B, A and O.  (DC seems fine.)

Would you consider changing the typename to BeforeFn and AfterFn, and changing the argument names to Before, After and OS?